src/org/parosproxy/paros/core/scanner/Alert.java

```diff
@@ -332,7 +337,33 @@ public int compareTo(Object o2) throws ClassCastException {
 			return result;
 		}
 		
-		return otherInfo.compareToIgnoreCase(alert2.otherInfo);
+		result = otherInfo.compareToIgnoreCase(alert2.otherInfo);
+		if (result != 0) {
+			return result;
+		}
+
+		if (evidence == null) {
+			if (alert2.evidence == null) {
+				return 0;

```

Comment:
Shouldnt we carry on here instead of returning 0? We still need to check the attack etc